pkg/pmem-csi-operator/controller/deployment/controller_driver.go

```diff
@@ -25,18 +26,52 @@ import (
 	rbacv1 "k8s.io/api/rbac/v1"
 	storagev1beta1 "k8s.io/api/storage/v1beta1"
 	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	apiruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	"k8s.io/klog"
+	"k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
 	controllerServicePort = 10000
 	controllerMetricsPort = 10010
 	nodeControllerPort    = 10001
+
+	// Annotation used to tag/track the version of the operator that
+	// successfully reconciled a deployment.
+	VersionAnnotation = "pmem-csi.intel.com/operatorVersion"
+
+	// Used by unit testing
+	TestVersion = "testing"
 )
 
+// A list of all object types potentially created by the
+// operator. It's okay and desirable to list more than actually used
+// at the moment, to catch new objects.
```

Comment:
This comment was applicable only when this was used for testing. It's not desirable to add unused objects anymore because that then would imply that we also add those to the RBAC rules.

Now a more accurate description is:

```
A list of all object types potentially created by the operator, in this or any previous release. In other words, this list may grow, but never shrink, because a newer release needs to delete objects created by an older release.

 This list also must be kept in syc with the operator RBAC rules.
```

Reply:
Rephrased the comment.